Description: Create batman_if only on register event
 We try to get all events for all net_devices to be able to add special sysfs
 folders for the batman-adv configuration. This also includes such events like
 NETDEV_POST_INIT which has no valid kobject according to
 v2.6.32-rc3-13-g7ffbe3f. This would create an oops in that situation.
 .
 It is enough to create the batman_if only on NETDEV_REGISTER events because we
 will also receive those events for devices which already existed when we
 registered the notifier call.
Author: Sven Eckelmann <sven.eckelmann@gmx.de>
Bug-Debian: http://bugs.debian.org/593471

---
diff --git a/hard-interface.c b/hard-interface.c
index b2c53b771e337b979f7cf7d0054a37778e176a43..67f5ed6ae76bb075fb400bc117d37b4e5658aa0a 100644
--- a/hard-interface.c
+++ b/hard-interface.c
@@ -398,15 +398,13 @@ static int hard_if_event(struct notifier_block *this,
 	/* FIXME: each batman_if will be attached to a softif */
 	struct bat_priv *bat_priv = netdev_priv(soft_device);
 
-	if (!batman_if)
-		batman_if = hardif_add_interface(net_dev);
+	if (!batman_if && event == NETDEV_REGISTER)
+			batman_if = hardif_add_interface(net_dev);
 
 	if (!batman_if)
 		goto out;
 
 	switch (event) {
-	case NETDEV_REGISTER:
-		break;
 	case NETDEV_UP:
 		hardif_activate_interface(bat_priv, batman_if);
 		break;
